gen/cycle.ml

```diff
@@ -709,25 +711,25 @@ let remove_store n0 =
 (* Set locations of events *)
 (***************************)
 
-  let is_read_same_nonfetch m =
+let is_read_same_fetch m =
     let check n = (n != m && (loc_compare n.evt.loc  m.evt.loc) = 0 && n.evt.dir = Some R &&
-                  not (E.is_ifetch n.edge.E.a1)) in
+                  (E.is_ifetch n.edge.E.a1)) in
     try ignore (find_node_prev (fun n -> check n) m); true
     with Not_found -> false
 
   let check_fetch n0 =
     let rec do_rec m =
       (* ensure Instr read is followed or preceded by plain read to same location*)
       begin match m.evt.loc, m.evt.dir with
-        | Code.Code _, Some R when not (E.is_ifetch m.edge.E.a1) ->
-            if is_read_same_nonfetch m then begin
+        | Code.Code _, Some R when (E.is_ifetch m.edge.E.a1) ->
+            if is_read_same_fetch m then begin
```

Comment:
I have doubts on the usefulness of that clause and of the next one. I cannot decide. If you cannot either, let us keep them.